skfp/model_selection/randomized_scaffold_split.py

```diff
@@ -0,0 +1,380 @@
+import warnings
+from collections import defaultdict
+from collections.abc import Sequence
+from numbers import Integral
+from typing import Any, Optional, Union
+
+import numpy as np
+from rdkit.Chem import Mol
+from rdkit.Chem.Scaffolds import MurckoScaffold
+from sklearn.utils._param_validation import Interval, RealNotInt, validate_params
+
+from skfp.model_selection.utils import (
+    ensure_nonempty_list,
+    get_data_from_indices,
+    split_additional_data,
+    validate_train_test_sizes,
+    validate_train_valid_test_split_sizes,
+)
+from skfp.utils.validators import ensure_mols
+
+
+@validate_params(
+    {
+        "data": ["array-like"],
+        "additional_data": ["tuple"],
+        "train_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "test_size": [
+            Interval(RealNotInt, 0, 1, closed="neither"),
+            Interval(Integral, 1, None, closed="left"),
+            None,
+        ],
+        "include_chirality": ["boolean"],
+        "use_csk": ["boolean"],
+        "return_indices": ["boolean"],
+        "random_state": [int, np.random.RandomState],
```

Comment:
Previous string `"random_state"` was good, do not change it, Scikit-learn validator will work properly